### PR TITLE
feat(btc): use new snap account flow for Bitcoin accounts

### PR DESCRIPTION
**Pull request by ccharly**
## **Description**

This removes the old flow of creating Bitcoin accounts in favor of the new generic Snap account creation flow (that includes account renaming).

This PR does require the Bitcoin Snap to emit the suggested names, otherwise name accounts will be named as follow: `Snap Account N` (where `N` is the number of created Snap accounts + 1).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26183?quickstart=1)

## **Related issues**

Relates to:
- https://github.com/MetaMask/snap-bitcoin-wallet/pull/210

Fixes:
- https://github.com/MetaMask/accounts-planning/issues/530

## **Manual testing steps**

> [!NOTE]
> Since we have updated the Bitcoin Snap version to `0.4.0`, you no longer need to manually install the Snap, it's being prebundled with the right version that supports account name suggestion

1. Create a Bitcoin account (should be named "Bitcoin Account" by default)
2. Create a Bitcoin testnet account (should be named "Bitcoin Testnet Account" by default)

### **After**

https://github.com/user-attachments/assets/5753ab25-ceb0-441b-a750-3c028a9dc9d8

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/confirmation/templates/create-named-snap-account.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ import {
  */
 function getValues(pendingApproval, t, actions, _history, _data, contexts) {
   const { origin: snapId, snapName, requestData } = pendingApproval;
-  const { address, snapSuggestedAccountName } = requestData;
+  const { snapSuggestedAccountName } = requestData;
```
[the remark this answers is not in the document]
Comment (ccharly) [reply ui/pages/confirmations/confirmation/templates/create-named-snap-account.js:21]: That's a good point! I'll remove that too, thanks! :)

[the remark this answers is not in the document]
Comment (ccharly) [reply ui/pages/confirmations/confirmation/templates/create-named-snap-account.js:21]: Done here: https://github.com/MetaMask/metamask-extension/pull/26183/commits/e5a7ec0f90552b776688fa55d0a35c96a912616e

**Review by k-g-j — APPROVED** [review]
Lgtm! 

**Manual testing**

Create Bitcoin mainnet account
https://github.com/user-attachments/assets/52349099-7207-4855-89ad-779dd5d2b328

Create Bitcoin testnet account
https://github.com/user-attachments/assets/fbf71c59-4afc-4848-97c1-625e0fa767a6

**Review by gantunesr — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Comment by k-g-j** [comment]
Behaves as epected on Chrome version 128.0.6613.138 ✅ 

https://github.com/user-attachments/assets/63c0b76c-8df1-4c98-930b-44a501422f2f

Behaves as expect on Firefox version 130.0 ✅ 

https://github.com/user-attachments/assets/e008c9e9-decf-415d-b899-1ebe9c7e1136

**Pull request merged**